src/wallet/api/wallet.cpp

```diff
@@ -2062,20 +2062,20 @@ bool WalletImpl::verifyMessageWithPublicKey(const std::string &message, const st
 
 bool WalletImpl::connectToDaemon()
 {
-    bool result = m_wallet->check_connection(NULL, NULL, DEFAULT_CONNECTION_TIMEOUT_MILLIS);
+    const expect<void> result = m_wallet->check_connection(NULL, NULL, DEFAULT_CONNECTION_TIMEOUT_MILLIS);
```

Comment:
@xiphon @selsta 

This patch stores additional information in the wallet API error message. Sometimes the message from OpenSSL in particular is a little funky, but the `error_code` typically provides more information as to why the `check_connection` attempt failed. Is there a better place for this?